### PR TITLE
Bookstore: connect to API

### PR DESCRIPTION
**Pull request by Mukumbuta**
### Feature implemented:
- Used [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) with` async/await` to make API calls.
- Fetched book list from Bookstore API on Books page load.
- Dispatched another action that passed API response to reducer and updates the state.
- Refactored the `add book` and `remove book` features to persist your changes in the server.

- Add book and remove book works in the same way after the refactor.

### CODE REVIEW

File: src/redux/books/book.js [hunk 1]
```diff
@@ -42,5 +23,61 @@ const booksReducer = (state = initState, action) => {
...
 
+const apiBooksFetched = (payload) => ({
+  type: API_BOOKS_FETCHED,
+  payload,
+});
+
+const addBookAction = (payload) => ({
+  type: BOOK_ADDED,
+  payload,
+});
```
Comment (fobadara) [inline src/redux/books/book.js]: [REQUIRED] Your addBook is not functioning properly.  It only displays a button unless we refresh.

|Your Bookstore|
|-----------------|
|bookstore1|

File: src/redux/configureStore.js [hunk 2]
```diff
@@ -1,12 +1,21 @@
-import { configureStore } from '@reduxjs/toolkit';
+// import { configureStore } from '@reduxjs/toolkit';
+import { combineReducers, createStore, applyMiddleware } from 'redux';
+import { composeWithDevTools } from 'redux-devtools-extension';
+import logger from 'redux-logger';
+import thunk from 'redux-thunk';
```
Comment (fobadara) [inline src/redux/configureStore.js]: [REQUIRED] You should be using createAsyncThunk from the Redux toolkit

**Review by fobadara — CHANGES_REQUESTED** [review]
Hi @Mukumbuta, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## Highlights
✅ Linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:

[OPTIONAL] You. don't need to add bookstore/  to your branch name. Just name it after a feature or action in the project.

[OPTIONAL] Kindly ensure that your commit messages are I the imperative and start with uppercase

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by victorokoroji — APPROVED** [review]
Hi @Mukumbuta, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
congratulations!  🎉💯🌟 

## Top highlight 🎯

- The Pull Request is following the GitHub Flow ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**